### DISCUSSION TITLE
Automatic text OCR to clipboard via selection

### DISCUSSION THREAD

**Discussion by ghost** [flameshot-org/flameshot#3677] [Ideas]
https://github.com/user-attachments/assets/92b793da-8d81-4346-bc29-45ca5f5d8d26

A screenshot ocr like this, I think it would be awesome and with how much AI has improved maybe it could be done. I could do a hotkey like ctrl+meta+c to copy the text I select. Like imagine if you look at a picture and see a restaurant name in the background but it is long so you do ctrl+meta+c select the area and boom you have it in your clipboard or in videos, if you want to reverse search a quote or something.

**Comment by ghost** [comment]
Alright I have figured out how to do it without needing to change the program at all. Here are the steps I took to get working OCR.

**Showcase**:

https://github.com/user-attachments/assets/30b183f0-df74-42e3-9796-93ba6334260d


**Installing xclip, tesseract and tesseract's training data, then moving its training data to the appropriate location** 

note: you can check where you need to move the files by running `tesseract`, the error will state where the files are missing. It will either be in `/usr/share` or `/usr/local`

Install the appropriate packages:
`sudo pacman -S tesseract xclip`
 
Clone the training data:
`wget https://github.com/tesseract-ocr/tessdata/raw/main/eng.traineddata`

Move the training data to tesseract:
`sudo mv eng.traineddata /usr/share/tessdata/` 

**Making a bash script to OCR a temporary flameshot image**

Make a shell script in whatever location you chose, in this case I will be using `~/Documents/Scripts`.

`nvim ~/Documents/Scripts/scocr.sh`

 Then paste the following code into it:

```
#!/bin/bash

flameshot gui --accept-on-select --path /tmp/screenshot.png

if [ -f /tmp/screenshot.png ]; then
   
    tesseract /tmp/screenshot.png /tmp/ocr_output

    cat /tmp/ocr_output.txt | xclip -selection clipboard

    rm /tmp/screenshot.png /tmp/ocr_output.txt
else
    echo "Screenshot capture failed."
fi
```

And finally, make the file executable by running 

`chmod +x scocr.sh`

**Making a hotkey for the script on KDE**

Open Settings>System Settings>Keyboard>Shortcuts and click 'Add New', select 'Command or Script'.

And type the following command: 

`sh /home/Username/ScriptLocation` 

Replace Username and ScriptLocation with your local user and the location with the location of the script.

Click on 'Add custom shortcut' and add whatever key combination you want.

Click 'Apply'

**You now have OCR with the press of a few buttons!**

**Conclusion**

I do not think that there is a need to add anything to the program, the user can perfectly integrate such functionality on their own. However if the community wishes to, they can integrate tesseract into the program to add it as an option within it. It would not consume much memory or space and could be a neat addition, which as to my knowledge, no other screenshot program has right now!

**Comment by mmahmoudian** [comment]
About three years ago I posted this which might help you improve your workflow:

https://mehrad.ai/posts/20210702-extracting-payment-info-in-rasterized-invoice/
